### DISCUSSION TITLE
Cd command doesn't work

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [community/community#26631] [Actions]
<p>Hello! I’m trying to write <a href="https://github.com/alvinseville7cf/System.AI/blob/fa900f22b9866c68bac38f3142302ec79c7fd80c/.github/workflows/ci.yml" rel="noopener nofollow ugc">CI</a> to compile Mono-based project:</p>
<pre><code class="lang-auto">name: CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

  workflow_dispatch:

jobs:
  build:
    runs-on: macos-latest

    steps:
      - uses: actions/checkout@v2

      - name: Mono build
        run: |
          cd Implementation
          chmod +x Compile.Mono.sh
          ./Compile.Mono.sh
</code></pre>
<p>But when I run it I obtain: <code>./Compile.Mono.sh: line 1: cd: System.AI/Implementation/: No such file or directory</code>. I don’t understand why it’s going on. <code>ls</code> command shows that there is Implementation directory…</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>There is an <code>Implementation/</code> directory in your repository, so the <code>cd Implementation</code> command in your <code>run</code> step works. The error message is from your <code>Compile.Mono.sh</code> script, which does another <code>cd System.AI/Implementation/</code> in its first line, and that doesn’t exist inside the <code>Implementation/</code> directory.</p>
<p>The best solution would probably be to make the script independent of where it is called from, e.g.:</p>
<pre><code class="lang-auto">cd "$(dirname ${0})"
mkdir Release
# ...
</code></pre>
<p>And remove the final <code>cd</code>, it doesn’t matter at all as long as you call (and not source) the script. :slightly_smiling_face:</p>
